.github/workflows/check-3rd-party.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+FILE=$1
+status=0
+
+for line in `sed -ne 's/[[:space:]-]*uses:[[:space:]]*//p' $1 | sed -e 's/\s*#.*$//'`; do
+  author=`echo $line | awk -F/ '{print $1}'`
+  if [ $author == "actions" ]; then continue; fi
+  version=`echo $line | awk -F@ '{print $2}' | awk '{print $1}'`
+  if [ ${#version} != 40 ]; then
```

Comment:
perhaps consider a regexp with [[, so that you don't accept a tag that just so happens to be 40 characters long: https://www.networkworld.com/article/2693361/unix-tip-using-bash-s-regular-expressions.html

You can never be sure, but at least you can require exactly 40 hexadecimal characters

Reply:
Makes sense.